fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -20,9 +20,20 @@
 reference = 23
 
 class C(object):
+    def foo(self): pass
     @staticmethod
-    def foo(): pass
-def bar(): pass
+    def foo_staticmethod(): pass
+    @classmethod
+    def foo_classmethod(cls): pass
+
+def fake(arg): pass
+def fake2(arg): pass
+def fake_no_args(): pass
+@staticmethod
```

Comment:
So despite my opinion on what cls and self should be bound to it seems that what's currently happening in practice is that they capture the class or instance that they're defined on. So your example assert will pass.